### PR TITLE
chore(deps): Hotfix: update Cocoa SDK from 8.29.1 to 8.36.0

### PR DESCRIPTION
**Pull request by antonis**
Bumps Cocoa SDK version from `8.29.1` to `8.36.0` for hotfix release `5.24.3`.
Also:
* removed duplicate `SentryTraceHeader` definition [as in 8.35.0 update](https://github.com/getsentry/sentry-react-native/pull/4037/commits/a218f44f4a70b544d026e9d465893afd8a56a16e)
* fixed sourceMapString is not a function, support metro@0.80.10 (see #4004)
* updated to RN 0.73.9, fix CI flipper error (see #3981)

### Features

- Continuous mode profiling (see `SentrySDK.startProfiler` and `SentryOptions.profilesSampleRate`) (#4010)

### Fixes

- Proper redact SR during animation (#4289)

### Fixes

- Crash when reading corrupted envelope (#4297)

### Features

- Expose span baggage API (#4207)

### Fixes

- Fix `SIGABRT` when modifying scope user (#4274)
- Crash during SDK initialization due to corrupted envelope (#4291)
  - Reverts [#4219](https://github.com/getsentry/sentry-cocoa/pull/4219) as potential fix

### Features

- Pause replay in session mode when offline (#4264)
- Add replay quality option for Objective-C (#4267)

### Fixes

- Session replay not redacting buttons and other non UILabel texts (#4277)
- Rarely reporting too long frame delays (#4278) by fixing a race condition in the frames tracking logic.
- Crash deserializing empty envelope length>0 (#4281]
- Guard dereferencing of stack frame pointer in SentryBacktrace ([#4268](https://github.com/getsentry/sentry-cocoa/pull/4268))

### Note: Due to a bug (#4280) introduced in this release, we recommend upgrading to [8.35.0](https://github.com/getsentry/sentry-cocoa/releases/tag/8.35.0) or newer.

---

This release fixes a bug (#4230) that we introduced with a refactoring (#4101) released in [8.30.1](https://github.com/getsentry/sentry-cocoa/releases/tag/8.30.1).
This bug caused unhandled/crash events to have the unhandled property and mach info missing, which is required for release health to show events in the unhandled tab. It's essential to mention that this bug **doesn't impact** release health statistics, such as crash-free session or user rates.

### Features

- Support orientation change for session replay (#4194)
- Replay for crashes (#4171)
- Redact web view from replay (#4203)
- Add beforeCaptureViewHierarchy callback (#4210)
- Rename session replay `errorSampleRate` property to `onErrorSampleRate` (#4218)
- Add options to redact or ignore view for Replay (#4228)

### Fixes

- Skip UI crumbs when target or sender is nil (#4211)
- Guard FramesTracker start and stop (#4224)
- Long-lasting TTID/TTFD spans (#4225). Avoid long TTID spans when the FrameTracker isn't running, which is the case when the app is in the background.
- Missing mach info for crash reports (#4230)
- Crash reports not generated on visionOS (#4229)
- Don’t force cast to `NSComparisonPredicate` in TERNARY operator (#4232)
- Fix accessing UI API on bg thread in enrichScope (#4245)
- EXC_BAD_ACCESS in SentryMetricProfiler (#4242)
- Missing '#include <sys/_types/_ucontext64.h>' (#4244)
- Rare flush timeout when called in tight loop (#4257)

### Improvements

- Reduce memory usage of storing envelopes (#4219)
- Skip enriching scope when nil (#4243)

### Features

- Add `reportAccessibilityIdentifier` option (#4183)
- Record dropped spans (#4172)

### Fixes

- Session replay crash when writing the replay (#4186)

### Features

- Collect only unique UIWindow references (#4159)

### Deprecated

- options.enableTracing was deprecated. Use options.tracesSampleRate or options.tracesSampler instead. (#4182)

### Fixes

- Session replay video duration from seconds to milliseconds (#4163)

### Features

- Include the screen names in the session replay (#4126)

### Fixes

- Properly handle invalid value for `NSUnderlyingErrorKey` (#4144)
- Session replay in buffer mode not working (#4160)

### Fixes

- UIKitless configurations now produce a module with a different name (#4140)
- Sentry Replay Serialized Breadcrumbs include level name ([#4141](https://github.com/getsentry/sentry-cocoa/pull/4141))

### Features

- Restart replay session with mobile session (#4085)
- Add pause and resume AppHangTracking API (#4077). You can now pause and resume app hang tracking with `SentrySDK.pauseAppHangTracking()` and `SentrySDK.resumeAppHangTracking()`.
- Add `beforeSendSpan` callback (#4095)

### Fixes

- `storeEnvelope` ends session for unhandled errors (#4073)
- Deprecate `SentryUser.segment`(#4092). Please remove usages of this property. We will remove it in the next major.
- Double-quoted include in framework header (#4115)
- Sentry Replay Network details should be available without Tracing (#4091)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## 5.24.3
```
Comment (kahest) [inline CHANGELOG.md:3]:
```suggestion
## Unreleased
```

Comment (kahest) [reply CHANGELOG.md:3]: triggering the actual release will update this

**Review by kahest — CHANGES_REQUESTED** [review]

**Review by kahest — APPROVED** [review]
LGTM, thanks!

**Comment by antonis** [comment]
Adding some notes on the failing CI tests:
* `Build RN 0.73.2 new hermes android production no` and `Build RN 0.73.2 legacy hermes android production no` should be related with [the sourceMapString is not a function fix](https://github.com/getsentry/sentry-react-native/pull/4004)
* `Build legacy ios dev no-frameworks`, `Build legacy ios production no-frameworks`, `Build new ios dev no-frameworks`, `Build new ios production no-frameworks`, `Build RN 0.73.2 legacy hermes ios production no`, `Build RN 0.73.2 new hermes ios production no` should be related with [the fix CI flipper error](https://github.com/getsentry/sentry-react-native/pull/3981)
* the `End-to-End Tests / metrics (legacy, android) (pull_request)`, `End-to-End Tests / metrics (new, ios) (pull_request)` is a known issue

**Review by lucas-zimerman — COMMENTED** [review]
Thank you for the PR!
The tests should be passing before merging, but other than that the PR looks good!

**Comment by lucas-zimerman** [comment]
@antonis I saw the following error messages on the CI tests:
```
error Your lockfile needs to be updated, but yarn was run with `--frozen-lockfile`.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

Could you try updating the branch locally and update (yarn install) the samples and the SDK project?

**Pull request merged**